### PR TITLE
[AC-2504] Add new members access report card

### PR DESCRIPTION
**Pull request by aj-rosado**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add new member access report card with a new guard validating if org is enterprise.

## Code changes

- **is-enterprise-org.guard.ts:** Guard that validates if the org has enterprise access, otherwise shows a dialog asking the user to upgrade to enterprise.
- **organization-reporting-routing.module.ts:** Added new route to member access report page.
- **organizations/reporting/reports-home.component.ts:** Adding the member access report entry to the `reportsArray` when the feature flag is active
- **reports.ts**: Added new member access report
- **report-variant.ts**: Added new variant `RequiresEnterprise`

### **Light Layout:**

**Enterprise**
enterprise reports light layout

**Family/Teams**
family/teams reports light layout

**Free**
free reports light layout

**Upgrade Dialog**
upgrade dialog member access light layout

### **Dark Layout:**
**Enterprise**
enterprise reports light layout

**Family/Teams**
family/teams reports dark layout

**Free**
free reports dark layout

**Upgrade Dialog**
upgrade dialog member access dark layout

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/web/src/app/admin-console/organizations/guards/is-enterprise-org.guard.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  ) {}
+
+  async canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    const memberAccessReport = await firstValueFrom(
+      this.configService.getFeatureFlag$(FeatureFlag.MemberAccessReport),
+    );
+
+    if (!memberAccessReport) {
+      return this.router.createUrlTree(["/"]);
+    }
```
Comment (djsmith85) [inline apps/web/src/app/admin-console/organizations/guards/is-enterprise-org.guard.ts:29]: I'd probably remove this check, as the guard itself could be used by others, regardless of this feature flag. You are also only showing the card if the flag is enabled, so no need to check it again.

File: apps/web/src/app/admin-console/organizations/reporting/reports-home.component.ts [hunk 2]
```diff
@@ -14,13 +17,20 @@ export class ReportsHomeComponent implements OnInit {
   reports$: Observable<ReportEntry[]>;
   homepage$: Observable<boolean>;
 
+  private memberAccessReport: boolean;
```
Comment (djsmith85) [inline apps/web/src/app/admin-console/organizations/reporting/reports-home.component.ts:20]: This should be renamed to make it clear that it's a flag showing if the report is accesible/visible

Suggestion: `isMemberAccessReportEnabled`or `memberAccessReportEnabled` if you don't want to prefix with is/can/has

File: apps/web/src/locales/en/messages.json [hunk 3]
```diff
@@ -8223,6 +8223,7 @@
       }
     }
   },
+
```
Comment (djsmith85) [inline apps/web/src/locales/en/messages.json:8226]: Please remove empty line

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/web/src/app/admin-console/organizations/guards/is-enterprise-org.guard.ts (hunk 1 above)
Comment (aj-rosado) [reply apps/web/src/app/admin-console/organizations/guards/is-enterprise-org.guard.ts:29]: I added this to protect against navigating directly to the url, despite not displaying the card would still be possible to navigate directly through the endpoint

File: apps/web/src/app/admin-console/organizations/reporting/organization-reporting-routing.module.ts [hunk 4]
```diff
@@ -79,6 +80,15 @@ const routes: Routes = [
             },
             canActivate: [IsPaidOrgGuard],
           },
+          {
```
[the remark this answers is not in the document]
Comment (aj-rosado) [reply apps/web/src/app/admin-console/organizations/reporting/organization-reporting-routing.module.ts:83]: Will remove and add it later with the `MemberAccessComponent`

**Review by djsmith85 — DISMISSED** [review]

File: apps/web/src/app/admin-console/organizations/guards/is-enterprise-org.guard.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    private dialogService: DialogService,
+    private configService: ConfigService,
+  ) {}
+
+  async canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    const isMemberAccessReportEnabled = await firstValueFrom(
+      this.configService.getFeatureFlag$(FeatureFlag.MemberAccessReport),
+    );
+
+    if (!isMemberAccessReportEnabled) {
```
Comment (vincentsalucci) [inline apps/web/src/app/admin-console/organizations/guards/is-enterprise-org.guard.ts:27]: 🎨 Let's make sure this conditional is removed during the feature flag clean-up.

**Review by vincentsalucci — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Pull request merged**